types.go

```diff
@@ -559,10 +570,15 @@ func (t *ResourceType) CompositeInitializers() [][]Parameter {
 	return t.Initializers
 }
 
-// EventType
+type CompositeTypeID struct {
+	Location   string
+	Identifier string
+}
```

Comment:
Do you think it would make sense to have a separate `LocationKind` field that is one of `script`, `transaction`, `account`? This would replace the `S.`, `T.`, `A.` prefixes.

If we're splitting the type into pieces, why not split it all the way?

Reply:
That's very good idea! I started with fields I needed for other tasks, counting on feedback like this from people with more knowledge.

Reply:
@psiemens @m4ksio do you have any preference for modelling this either as 
- a "tagged struct": define a struct `TypeID`, which has a `LocationKind` field, and a constant for each location kind
- or as an interface `TypeID`, and dedicated structs for each location kind, e.g. `ScriptTypeID`, `AccountTypeID`, etc. (how e.g. `ast.Location` is declared at the moment)

Reply:
@turbolent I think I prefer a simple struct in this case, therefore no reflection required.

Reply:
Agree, I don't think we need more polymorphism here